File: backend/app/schemas/notification.py
from pydantic import BaseModel


class NotificationResponse(BaseModel):
    id: str
    type: str
    title: str
    message: str
    case_id: int
    case_no: str
    applicant: str | None = None
    executive: str | None = None
    occurred_at: str | None = None
    due_at: str | None = None
    severity: str

File: frontend/src/types/notification.ts
export type NotificationType =
  | "OVERDUE_FOLLOW_UP"
  | "TODAY_FOLLOW_UP"
  | "OLD_PENDING_CASE";

export type NotificationSeverity = "info" | "warning" | "critical";

export interface NotificationItem {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  case_id: number;
  case_no: string;
  applicant: string | null;
  executive: string | null;
  occurred_at: string | null;
  due_at: string | null;
  severity: NotificationSeverity;
}
